### PR TITLE
Adds QC Filtering to OOI Preload

### PR DESCRIPTION
**Pull request by lukecampbell**
Adds new parameter to `IONLoader`: `parseooi`

If you want to have `IONLoader` load the OOI Assets (SAF) into memory and have them be available to preload categories specify `parseooi=True` on the command line or in the configurations.

### CODE REVIEW

File: ion/processes/bootstrap/ion_loader.py [hunk 1]
```diff
@@ -1881,9 +1883,31 @@ def _load_ParameterDefs(self, row):
...
+                        'Combine QC Flags (CMBNFLG) QC'                          : 'cmbnflg_qc',
+                        }
+
+
+                qc_fields = None
+                if getattr(self.ooi_loader, 'ooi_objects', None):
+                    dps = self.ooi_loader.get_type_assets('data_product')
+                    if context.ooi_short_name in dps:
+                        dp = dps[context.ooi_short_name]
+                        qc_fields = [v for k,v in qc_map.iteritems() if dp[k] == 'applicable']
```
Comment (lukecampbell) [inline ion/processes/bootstrap/ion_loader.py:1908]: I'm not 100% on the semantic vocabulary of the SAF database but the only affirmative word I found was `applicable` the others were TBD, not-applicable, na.

**Comment by mmeisinger** [comment]
Excellent! :rocket: 

One small line comment attached

File: ion/processes/bootstrap/ion_loader.py [hunk 2]
```diff
@@ -1881,9 +1883,31 @@ def _load_ParameterDefs(self, row):
...
+                        'Evaluate Polynomial (POLYVAL) QC'                       : 'polyval_qc',
+                        'Solar Elevation (SOLAREL) QC'                           : 'solarel_qc',
+                        'Conductivity Compressibility Compensation (CONDCMP) QC' : 'condcmp_qc',
+                        '1-D Interpolation (INTERP1) QC'                         : 'interp1_qc',
+                        'Combine QC Flags (CMBNFLG) QC'                          : 'cmbnflg_qc',
+                        }
+
+
+                qc_fields = None
+                if getattr(self.ooi_loader, 'ooi_objects', None):
```
Comment (mmeisinger) [inline ion/processes/bootstrap/ion_loader.py]: The correct way of querying this is:
if self.ooi_loader._extracted:
   # Yes OOI assets were parsed

self.ooi_loader is always existing but by default self.ooi_loader._extracted == False

File: ion/processes/bootstrap/ion_loader.py (hunk 1 above)
Comment (mmeisinger) [inline ion/processes/bootstrap/ion_loader.py:1908]: This was an issue, but it seems "applicable" is the correct term to check for and treat everything else as not applicable. If this is somewhere messed up, the SAF data will need to be fixed

**Comment by blazetopher** [comment]
:fireworks:

**Pull request merged**